sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/WorkDirectoriesInitializer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.scanner.scan;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Iterator;
+import org.sonar.api.batch.fs.internal.DefaultInputModule;
+import org.sonar.api.batch.fs.internal.InputModuleHierarchy;
+import org.sonar.core.util.FileUtils;
+import org.sonar.home.cache.DirectoryLock;
+
+/**
+ * Clean and create working directories of each module.
+ * Be careful that sub module work dir might be nested in parent working directory.
+ */
+public class WorkDirectoriesInitializer {
+
+  private InputModuleHierarchy moduleHierarchy;
+
+  public WorkDirectoriesInitializer(InputModuleHierarchy moduleHierarchy) {
+    this.moduleHierarchy = moduleHierarchy;
+  }
+
+  public void execute() {
+    cleanAllWorkingDirs(moduleHierarchy.root());
+    mkdirsAllWorkingDirs(moduleHierarchy.root());
+  }
+
+  private void cleanAllWorkingDirs(DefaultInputModule module) {
+    for (DefaultInputModule sub : moduleHierarchy.children(module)) {
+      cleanAllWorkingDirs(sub);
+    }
+    if (Files.exists(module.getWorkDir())) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this with a call to the "toFile().exists()" method [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3725)